pkg/terra/client/test_helpers.go

```diff
@@ -75,49 +78,59 @@ func SetupLocalTerraNode(t *testing.T, chainID string) ([]Account, string) {
 		privateKey, address := createKeyFromMnemonic(t, k.Mnemonic)
 		require.Equal(t, expAcctAddr, address)
 		// Give it 100 luna
-		_, err = exec.Command("terrad", "add-genesis-account", k.Address, "100000000uluna", "--home", testdir).Output()
-		require.NoError(t, err)
+		out2, err2 := exec.Command("terrad", "add-genesis-account", k.Address, "100000000uluna", "--home", testdir).Output() //nolint:gosec
+		require.NoError(t, err2, string(out2))
 		accounts = append(accounts, Account{
 			Name:       account,
 			Address:    address,
 			PrivateKey: privateKey,
 		})
 	}
 	// Stake 10 luna in first acct
-	out, err := exec.Command("terrad", "gentx", accounts[0].Name, "10000000uluna", fmt.Sprintf("--chain-id=%s", chainID), "--keyring-backend", "test", "--keyring-dir", testdir, "--home", testdir).CombinedOutput()
+	out, err = exec.Command("terrad", "gentx", accounts[0].Name, "10000000uluna", fmt.Sprintf("--chain-id=%s", chainID), "--keyring-backend", "test", "--keyring-dir", testdir, "--home", testdir).CombinedOutput() //nolint:gosec
 	require.NoError(t, err, string(out))
 	out, err = exec.Command("terrad", "collect-gentxs", "--home", testdir).CombinedOutput()
 	require.NoError(t, err, string(out))
-	cmd := exec.Command("terrad", "start", "--home", testdir)
+	port := mustRandomPort()
+	tendermintURL := fmt.Sprintf("http://127.0.0.1:%d", port)
+	t.Log(tendermintURL)
+	cmd := exec.Command("terrad", "start", "--home", testdir, "--rpc.laddr", fmt.Sprintf("tcp://127.0.0.1:%d", port))
 	require.NoError(t, cmd.Start())
 	t.Cleanup(func() {
 		require.NoError(t, cmd.Process.Kill())
 	})
 	// Wait for api server to boot
 	var ready bool
 	for i := 0; i < 10; i++ {
-		time.Sleep(1 * time.Second)
-		out, err = exec.Command("curl", "http://127.0.0.1:26657/abci_info").Output()
-		require.NoError(t, err)
+		time.Sleep(time.Second)
+		out, err = exec.Command("curl", tendermintURL+"/abci_info").Output() //nolint:gosec
+		if err != nil {
+			t.Logf("API server not ready yet (attempt %d): %v\n", i+1, err)
+			continue
+		}
```

Comment:
This require was a bit too strict, since calling 'too soon' returns an exit code 7.